connectors/handlers/tasks.py

```diff
@@ -328,3 +331,126 @@ def slack_bot_handle_receive_message(slack_connector_id, message):
+        if not pagerduty_connector:
+            print(f"Error while handling PagerDuty handle_receive_message: Connector not found for connector_id: "
+                  f"{pagerduty_connector_id}")
+            return
+        pagerduty_connector = pagerduty_connector.first()
+        account_id = pagerduty_connector.account_id
+
+        bot_auth_token = get_db_connector_keys(account_id=account_id, connector_id=pagerduty_connector.id,
+                                               key_type=SourceKeyType.PAGER_DUTY_API_KEY)
+        if not bot_auth_token:
+            print(f"Error while handling PagerDuty handle_receive_message: Bot auth token not found for connector_id: "
+                  f"{pagerduty_connector_id}")
+            return
+        event_data = event.get('event', {}).get('data', {})
+        incident_id = event_data.get('id')
+        title = event_data.get('title')
+        incident_created_at = event_data.get('created_at')
+        # alerts = incident.get('alerts', [])
+        # alert_id = alerts.get('id')
+        # alert_text = alerts.get('summary')
+        # details = alerts.get('body', {}).get('details', 'No Details')
+        # alerts_created_at = alerts.get('created_at')
+
+        if not incident_created_at:
+            data_timestamp = datetime.utcnow()
+        else:
+            data_timestamp = datetime.strptime(incident_created_at, '%Y-%m-%dT%H:%M:%S%z')
+
+        if not incident_id:
+            print(f"Incident ID not found, ignoring incident: {incident_id}")
+            return
+
+        pagerduty_received_msg = PagerDutyConnectorDataReceived(
+            account_id=pagerduty_connector.account_id,
+            connector=pagerduty_connector,
+            incident_id=incident_id,
+            title=title,
+            incident_created_at=incident_created_at
+            # alert_id=alert_id,
+            # alert_text=alert_text,
+            # details=details,
+            # data_timestamp=data_timestamp
+        )
+        pagerduty_received_msg.save()
+
+        #change this line
+        all_alert_type_entry_points = get_db_workflow_entry_points(account_id=account_id,
+                                                                   entry_point_type=WorkflowEntryPointProto.ALERT)
+        ep_protos = [e.proto for e in all_alert_type_entry_points]
+
+        try:
+            entry_point_evaluator = get_entry_point_evaluator(WorkflowEntryPointProto.Type.ALERT)
+        except Exception as e:
+            print(f"Error while handling pagerduty_alert_trigger_playbook with error: {e} for event: {event} "
+                  f"for account: {account_id}")
+            return
+
+        for ep in ep_protos:
+            alert_config = ep.alert_config
+            if alert_config.alert_type == WorkflowEntryPointAlertConfigProto.AlertType.PAGERDUTY_INCIDENT_ALERT:
+                pagerduty_alert = {
+                    'incident_id': incident_id,
+                    'title': title
+                }
```

Comment:
Are you receiving service in incident details. Include them here if you are getting that detail.